esp/esdllib/README.md

```diff
@@ -0,0 +1,1608 @@
+
+Order | Transaction Step                                 | Script Entry Point
+------|--------------------------------------------------|-------------------
+  1   | Reciept of request from client                   |
+  2   | Authentication and Authorization                 |
+  3   | ESDL Transforms ESP Request to backend request   |
+  4   | | **BackendRequest** : Transform backend request to final form
+  5   | Backend is called, sending backend request       | **Service** : A script is run instead of calling out to the backend
+  6   | Backend response received                        |
+  7   | | **BackendResponse** : Transform backend response
+  8   | ESDL Transforms backend response to initial ESP response |
+  9   | | **ESDL Response** : Transform initial ESP response to final ESP response
+  10  | ESDL Transforms final ESP response to match schema |
+  11  | ESP sends final response to client |
+  12  | | **PreLogging** : Transform Log Request
+  13  | Logging Manager dispatches Log Request
+
+A script is inserted to an entry point by naming its root node after the name of the entry point. A script can replace a backend service call by configuring it as a **Service** entry point.
+
+Additional details about the entry points and script configuration is found in the reference section.
+
+### Illustrated by Flowchart
+
+The lifecycle of an ESP transaction is shown below in the flowchart's blue boxes. Optionally, scripts can be inserted at any of the entry points named in green boxes. At each entry point the script has the opportunity to transform the source in preparation for passing it on to the next transaction step.
+
+![ESP Transaction Flowchart with ESDL Script Entry Points](readme_assets/ESDLScriptEntryPoints.png)
+
+A script is inserted to an entry point by naming its root node after the name of the entry point. A script can replace a backend service call by configuring it as a **Service** entry point.
+
+Additional details about the entry points and script configuration is found in the reference section.
+
+## Scripting Context
+
+The scripting engine maintains a context during the lifetime of each transaction. This context holds state for the scripts, and the results of executing the operations at each entry point. The context contains an XML document holding most of the state that script operations can inspect and modify.
+
+Below is a first-level outline of the common sections in the XML context:
+
+*Verify these sections, they may have changed*
+
+```xml
+<esdl_script_context>
+    <esdl/>
+    <original_request/>
+    <target/>
+    <config/>
+    <esdl_request/>
+    <final_request/>
+    <initial_response/>
+    <pre_esdl_response/>
+    <logdata/>
+    <logging/>
+    <store/>
+    <temporaries/>
+</esdl_script_context>
+```
+
+Context Node | Contents
+-------------|-------------------
+`<esdl_script_context>` | Root node, not accessible by scripts
```

Comment:
For now instead of "not accessible by scripts" maybe say "should not be accessed by scripts".

I've thought about trying to make it not work, but haven't.

Reply:
Ok I thought it was impossible, but haven't actually tried. I'll switch to discourage.